src/easy/handler.rs

```diff
@@ -2049,7 +2050,14 @@ impl<H> Easy2<H> {
     /// `CURLOPT_CONNECTTIMEOUT_MS`.
     pub fn connect_timeout(&mut self, timeout: Duration) -> Result<(), Error> {
         let ms = timeout.as_secs() * 1000 + timeout.subsec_millis() as u64;
-        self.setopt_long(curl_sys::CURLOPT_CONNECTTIMEOUT_MS, ms as c_long)
+
+        match c_long::try_from(ms) {
+            Ok(amt) => self.setopt_long(curl_sys::CURLOPT_CONNECTTIMEOUT_MS, amt),
+            Err(_) => {
+                let amt = c_long::try_from(ms / 1000).unwrap();
```

Comment:
On second thought (after commenting on https://github.com/alexcrichton/curl-rust/pull/495), could this `.unwrap()` be translated to an error to be returned here instead? That way it's possible for callers to be notified that timeouts are simply too large.

Reply:
I tried to search the best error code to handle this for curl, but I cannot find that is related to overflow errors.

Reply:
Perhaps `CURLE_BAD_FUNCTION_ARGUMENT`?

Reply:
Please check, I added a Unit Test too